pkg/operator/controller/garden/garden/reconciler_delete.go

```diff
@@ -357,13 +372,17 @@ func (r *Reconciler) delete(
 	return reconcile.Result{}, nil
 }
 
-func (r *Reconciler) checkIfVirtualGardenManagedResourcesExist() func(context.Context) error {
+func (r *Reconciler) checkIfNonCleanupRelevantVirtualGardenManagedResourcesExist() func(context.Context) error {
```

Comment:
```suggestion
func (r *Reconciler) checkIfOnlyCleanupRelevantVirtualGardenManagedResourcesExist() func(context.Context) error {
```

Nit: Isn't that more matching? Because if there is at least one MR, an error will be returned.